File: labs_26_04_18/ex1/Circle.java
package labs_26_04_18.ex1;

// Klasa dla koła
public class Circle extends Figure {
    private final double radius;

    public Circle(double radius) {
        if (radius <= 0)
            throw new IllegalArgumentException("Promień musi być większy od zera.");
        this.radius = radius;
    }

    @Override
    public double calculateArea() {
        return Math.PI * radius * radius;
    }

    @Override
    public double calculatePerimeter() {
        return 2 * Math.PI * radius;
    }

    @Override
    public String getName() {
        return "Koło";
    }
}

File: labs_26_04_18/ex2/src/figures/circle.cpp
#include "../../inc/circle.hpp"
#include <cmath>
#include <stdexcept>

Circle::Circle(double r) : radius(r) {
  if (r <= 0)
    throw std::invalid_argument("Promien musi byc wiekszy od zera.");
}
double Circle::calculateArea() const {
  return std::acos(-1.0) * radius * radius;
}
double Circle::calculatePerimeter() const {
  return 2 * std::acos(-1.0) * radius;
}
std::string Circle::getName() const { return "Kolo"; }